tensorflow/lite/micro/micro_profiler.cc

```diff
@@ -67,4 +67,58 @@ void MicroProfiler::LogCsv() const {
 #endif
 }
 
+void MicroProfiler::LogTicksPerTagCsv() {
+#if !defined(TF_LITE_STRIP_ERROR_STRINGS)
+  MicroPrintf(
+      "\"Unique Ops in the Graph\",\"Total Ticks (all instances of the Op)\"");
+  int totalTicks = 0;
+  for (int i = 0; i < num_events_; ++i) {
+    uint32_t ticks = end_ticks_[i] - start_ticks_[i];
+    int position = FindExistingOrNextPosition(tags_[i]);
+    total_ticks_per_tag[position].tag = tags_[i];
+    total_ticks_per_tag[position].ticks =
+        total_ticks_per_tag[position].ticks + ticks;
+    totalTicks += ticks;
+  }
+
+  for (int i = 0; i < num_events_; ++i) {
+    ticks_per_tag ticksPerTag = total_ticks_per_tag[i];
+    if (ticksPerTag.tag == nullptr) {
+      break;
+    }
+    MicroPrintf("%s,%d", ticksPerTag.tag, ticksPerTag.ticks);
+  }
+  MicroPrintf("total,%d", totalTicks);
+
+#endif
+}
+
+int MicroProfiler::FindExistingOrNextPosition(const char* tagName) {
```

Comment:
Could we have some comments on what this function does?

Reply:
Thank you for the suggestion Pauline!  Adding a descriptive comment.